pkg/validation-webhook/snapshot.go

```diff
@@ -97,12 +126,35 @@ func decideSnapshot(snapshot, oldSnapshot *volumesnapshotv1beta1.VolumeSnapshot,
 		// Which allows the remover of finalizers and therefore deletion of this object
 		// Don't rely on the pointers to be nil, because the deserialization method will convert it to
 		// The empty struct value. Instead check the operation type.
-		if err := utils.ValidateSnapshot(oldSnapshot); err != nil {
+		if err := utils.ValidateV1Beta1Snapshot(oldSnapshot); err != nil {
+			return reviewResponse
+		}
+
+		// if it is an UPDATE and oldSnapshot is valid, check immutable fields
+		if err := checkSnapshotImmutableFieldsV1beta1(snapshot, oldSnapshot); err != nil {
+			reviewResponse.Allowed = false
+			reviewResponse.Result.Message = err.Error()
 			return reviewResponse
 		}
+	}
+	// Enforce strict validation for CREATE requests. Immutable checks don't apply for CREATE requests.
+	// Enforce strict validation for UPDATE requests where old is valid and passes immutability check.
+	if err := utils.ValidateV1Beta1Snapshot(snapshot); err != nil {
+		reviewResponse.Allowed = false
+		reviewResponse.Result.Message = err.Error()
+	}
+	return reviewResponse
+}
 
+func decideSnapshotV1(snapshot, oldSnapshot *volumesnapshotv1.VolumeSnapshot, isUpdate bool) *v1.AdmissionResponse {
```

Comment:
nit, either we remove "v1" for all v1 resource related functions or variables, or append "v1" for all to make it consistent. i.e., ValidateSnapshot